tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/Lambda.cs

```diff
@@ -0,0 +1,83 @@
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="incommingEvent">IncommingEvent value</param>
+        /// <param name="context">Context value.</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, object incommingEvent, object context)
+        {
+            Log.Debug("OnMethodBegin");
+            return new CallTargetState(CreatePlaceholderScope(TraceContextEndpoint, Tracer.Instance));
+        }
+
+        /// <summary>
+        /// OnMethodEnd callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TReturn">Type of the response, in an async scenario will be T of Task of T</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="returnValue">HttpResponse message instance</param>
+        /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+        /// <param name="state">Calltarget state value</param>
+        /// <returns>A response value</returns>
+        internal static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, in CallTargetState state)
+        {
+            Log.Debug("OnMethodEnd");
+            state.Scope?.Dispose();
+            return new CallTargetReturn<TReturn>(returnValue);
+        }
+
+        internal static Scope CreatePlaceholderScope(string traceContextEndpoint, Tracer tracer)
+        {
+            Scope scope = null;
+            try
+            {
+                WebRequest request = WebRequest.Create(traceContextEndpoint);
+                request.Credentials = CredentialCache.DefaultCredentials;
+                HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+
+                string placeholderServiceName = "placeholder-service";
+                string placeholderOperationName = "placeholder-operation";
+                string traceId = response.Headers.Get(HttpHeaderNames.TraceId);
+                Log.Debug("trace-id received: " + traceId);
+                // need to set the exact same spanId so nested spans (auto-instrumentation or manual) will have the correct parent-id
+                string spanId = response.Headers.Get(SpanIdHeader);
+                Log.Debug("spanId-id received: " + spanId);
+                SpanContext context = tracer.CreateSpanContext(null, null, false, Convert.ToUInt64(traceId), null);
+                var span = tracer.StartSpan(placeholderOperationName, null, context, placeholderServiceName, null, false, null, Convert.ToUInt64(spanId), true);
+                scope = tracer.TracerManager.ScopeManager.Activate(span, true);
```

Comment:
If I understand correctly, you're trying to create a span here with the same TraceId / SpanId as the one at the endpoint? If so, I think you can simplify the code in the following ways:

1. Keep only the optional arguments where you are setting non-default values:

```csharp
var span = tracer.StartSpan(placeholderOperationName, null, parent: context, serviceName: placeholderServiceName, spanId: Convert.ToUInt64(spanId));
```

2. In the StartSpan call, pass the traceId instead of the context you created from CreateSpanContext. This will have the same result but avoids creating an additional SpanContext. After applying both suggestions, the result should be this:

```csharp
// Delete call to tracer.CreateSpancontext
var span = tracer.StartSpan(placeholderOperationName, null, serviceName: placeholderServiceName, traceId: Convert.ToUInt64(traceId), spanId: Convert.ToUInt64(spanId));
```

Reply:
Thanks ! ✅